tikv/raftstore/fsm_apply.go

```diff
@@ -136,7 +186,69 @@ type notifier struct {
 	router *router
 }
 
+type applyContext struct {
+	tag              string
+	timer            *util.SlowTimer
+	host             *CoprocessorHost
+	router           applyRouter
+	notifier         notifier
+	engines          *Engines
+	cbs              util.MustConsumerVec
+	applyTaskResList []ApplyTaskRes
+	execCtx          *applyExecContext
+	wb               *WriteBatch
+	wbLastBytes      uint64
+	wbLastKeys       uint64
+	lastAppliedIndex uint64
+	committedCount   uint64
+
+	enableSyncLog  bool
+	syncLogHint    bool
+	useDeleteRange bool
+}
+
+type waitSourceMergeState struct {
+	pendingEntries []*eraftpb.Entry
+	pendingMsgs    []Msg
+	readyToMerge   *atomic.Uint64
+	catchUpLogs    *catchUpLogs
+}
+
+type applyDelegate struct {
+	id                uint64
+	term              uint64
+	region            *metapb.Region
+	tag               string
+	stopped           bool
+	pendingRemove     bool
+	pendingCmds       pendingCmdQueue
+	merged            bool
+	isMerging         bool
+	lastMergeVersion  uint64
+	waitMergeState    *waitSourceMergeState
+	readySourceRegion uint64
+	applyState        *rspb.RaftApplyState
+	appliedIndexTerm  uint64
+	metrics           ApplyMetrics
+}
+
+type applyDestroy struct {
+	regionId uint64
+}
+
+type catchUpLogs struct {
+	targetMailBox mailbox
+	merge         *raft_cmdpb.CommitMergeRequest
+	readyToMerge  *atomic.Uint64
+}
+
 type applyPollerBuilder struct {
+	tag             string
+	cfg             *Config
+	coprocessorHost *CoprocessorHost
+	engines         *Engines
+	sender          notifier
+	router          applyRouter
```

Comment:
use pointer here because applyRouter embeds router which is a big struct.

Reply:
ok .